src/helpers/ui/filter_interactive.ts

```diff
@@ -17,3 +17,58 @@ export function interactiveAddFilter(env: SpreadsheetChildEnv, sheetId: UID, tar
     env.raiseError(AddFilterInteractiveContent.nonContinuousTargets);
   }
 }
+
+export function adjecentCellsSelection(
```

Comment:
There is something that exists which does the same better: selectTableAroundSelection from selection stream processor. In addition to expand the zone, it also expand the selection to that expanded zone, which will approach the behaviour of GSheet.
Use it, directly in the function `interactiveAddFilter`.
You can check the action `CREATE_CHART` to see an exemple.